src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -253,72 +265,52 @@ private static bool ContainsNonAsciiByte_AdvSimd(uint advSimdIndex)
             return advSimdIndex < 16;
         }
 
-        private static unsafe nuint GetIndexOfFirstNonAsciiByte_Intrinsified(byte* pBuffer, nuint bufferLength)
+        private static unsafe nuint GetIndexOfFirstNonAsciiByte_Avx2(byte* pBuffer, nuint bufferLength)
         {
             // JIT turns the below into constants
 
-            uint SizeOfVector128 = (uint)Unsafe.SizeOf<Vector128<byte>>();
-            nuint MaskOfAllBitsInVector128 = (nuint)(SizeOfVector128 - 1);
+            uint SizeOfVector256 = (uint)Unsafe.SizeOf<Vector256<byte>>();
```

Comment:
`Unsafe.SizeOf` or `Vector256<byte>.Count` are going to be constants. It's likely better to use the values directly than to cache them in a local.

Reply:
-or- declare it as a proper `const uint SizeOfVector256 = 32` and then just assert its correct, which will give smaller IL and make the JIT do less work.